opm/simulators/flow/HybridNewtonConfig.cpp

```diff
@@ -0,0 +1,167 @@
+*/
+
+#include <config.h>
+#include <opm/simulators/flow/HybridNewtonConfig.hpp>
+
+#include <opm/common/ErrorMacros.hpp>
+
+#include <opm/simulators/linalg/PropertyTree.hpp>
+
+#include <algorithm>
+#include <fstream>
+#include <stdexcept>
+
+namespace Opm {
+
+HybridNewtonConfig::HybridNewtonConfig(const PropertyTree& model_config)
+{
+    model_path = model_config.get<std::string>("model_path", "");
+    if (model_path.empty()) {
+        throw std::runtime_error("Missing 'model_path' in HybridNewton config");
+    }
+
+    cell_indices_file = model_config.get<std::string>("cell_indices_file", "");
+    if (cell_indices_file.empty()) {
+        throw std::runtime_error("Missing 'cell_indices_file' in HybridNewton config");
+    }
+
+    // Load cell indices immediately
+    cell_indices = loadCellIndicesFromFile(cell_indices_file);
+    n_cells = cell_indices.size();
+
+    // Load apply times
+    auto applyTimesOpt = model_config.get_child_optional("apply_times");
+    if (!applyTimesOpt) {
+        throw std::runtime_error("Missing 'apply_times' in HybridNewton config");
+    }
+
+    for (const auto& key : applyTimesOpt->get_child_keys()) {
+        apply_times.push_back(applyTimesOpt->get_child(key).get<double>(""));
+    }
+
+    if (apply_times.empty()) {
+        throw std::runtime_error("'apply_times' must contain at least one value");
+    }
+
+    // Parse features
+    parseFeatures(model_config, "features.inputs", input_features);
+    parseFeatures(model_config, "features.outputs", output_features);
+}
+
+bool HybridNewtonConfig::hasInputFeature(const std::string& name) const
+{
+    return std::any_of(input_features.begin(), input_features.end(),
+                       [&](const auto& p) { return p.first == name; });
+}
+
+bool HybridNewtonConfig::hasOutputFeature(const std::string& name) const
+{
+    return std::any_of(output_features.begin(), output_features.end(),
+                       [&](const auto& p) { return p.first == name; });
+}
+
+void HybridNewtonConfig::validateConfig(bool compositionSwitchEnabled) const
+{
+    bool hasRsFeature = hasInputFeature("RS") ||
+                        hasOutputFeature("RS") ||
+                        hasOutputFeature("DELTA_RS");
+
+    bool hasRvFeature = hasInputFeature("RV") ||
+                        hasOutputFeature("RV") ||
+                        hasOutputFeature("DELTA_RV");
+
+    if ((hasRsFeature || hasRvFeature) && !compositionSwitchEnabled) {
+        OPM_THROW(std::runtime_error,
+            "HybridNewton: RS or RV features detected but composition support is disabled. "
+            "CompositionSwitch must be enabled for RS/RV features."
+        );
+    }
+}
+
+std::vector<int>
+HybridNewtonConfig::loadCellIndicesFromFile(const std::string& filename) const
+{
+    std::vector<int> indices;
+    std::ifstream cellFile(filename);
+    if (!cellFile.is_open()) {
+        throw std::runtime_error("Cannot open cell indices file: " + filename);
+    }
```

Comment:
I realise you're "just" moving some stuff around, and I appreciate that, but this situation cannot happen.  The `ifstream` constructor above will either successfully open the file or it will throw an exception. Whether you'd like to address this now or in follow-up work is up to you.